src/MPQEditor/MPQCircleSelector.ts

```diff
@@ -0,0 +1,319 @@
+    );
+  }
+
+  public static createOrShow(
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdOpenArgs,
+    handler: MPQSelectionEvent | undefined
+  ) {
+    let column = args.viewColumn;
+    if (column) {
+      if (column >= vscode.ViewColumn.One) {
+        column = column + 1;
+      }
+    }
+
+    // search for existing panel
+    const oldPanel = MPQSelectionPanel.findSelPanel(
+      args.modelPath,
+      args.document.uri.toString()
+    );
+    if (oldPanel) {
+      oldPanel._panel.reveal(column);
+      return;
+    }
+
+    // Otherwise, create a new panel.
+    const lastSlash = args.modelPath.lastIndexOf(path.sep) + 1;
+    const fileNameExt = args.modelPath.substring(lastSlash);
+    const panel = vscode.window.createWebviewPanel(
+      MPQSelectionPanel.viewType,
+      fileNameExt,
+      column || vscode.ViewColumn.Two,
+      { retainContextWhenHidden: true }
+    );
+
+    const graphSelPanel = new MPQSelectionPanel(
+      panel,
+      extensionUri,
+      args,
+      handler
+    );
+
+    MPQSelectionPanel.panels.push(graphSelPanel);
+    graphSelPanel.loadContent();
+    graphSelPanel.onForwardSelection(args.names);
+  }
+
+  private static findSelPanel(
+    docPath: string,
+    id: string
+  ): MPQSelectionPanel | undefined {
+    let result = undefined;
+    MPQSelectionPanel.panels.forEach((selpan) => {
+      if (docPath === selpan._modelPath && id === selpan._ownerId) {
+        result = selpan;
+      }
+    });
+    return result;
+  }
+
+  /**
+   * @brief called when owner is closing
+   */
+  public static closeByOwner(
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdCloseArgs
+  ) {
+    let selPanel = MPQSelectionPanel.findSelPanel(
+      args.modelPath,
+      args.document.uri.toString()
+    );
+    if (selPanel) {
+      selPanel.dispose();
+    }
+  }
+
+  /**
+   * @brief called when owner loads visq file
+   */
+  public static loadVisqInfo(
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdVisqArgs
+  ) {
+    let selPanel = MPQSelectionPanel.findSelPanel(
+      args.modelPath,
+      args.document.uri.toString()
+    );
+    if (selPanel) {
+      selPanel.onLoadVisqInfo(args.visqPath);
+    }
+  }
+
+  /**
+   * @brief called when owner selection state of nodes has changed
+   */
+  public static forwardSelectionByOwner(
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdLayersChangedArgs
+  ) {
+    let selPanel = MPQSelectionPanel.findSelPanel(
+      args.modelPath,
+      args.document.uri.toString()
+    );
+    if (selPanel) {
+      selPanel.onForwardSelection(args.names);
+    }
+  }
+
+  private constructor(
+    panel: vscode.WebviewPanel,
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdOpenArgs,
+    handler: MPQSelectionEvent | undefined
+  ) {
+    super(extensionUri, panel.webview);
+
+    // let parsedPath = path.parse(args.docPath);
+    // let fileBase = path.join(parsedPath.dir, parsedPath.name);
```

Comment:
Could it be removed, if it's not used?

Reply:
Ahhh.. Sorry. These lines will be removed.